### PR TITLE
#6 task.andrei.manya

### PR DESCRIPTION
**Pull request by AndreiManya**

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -46,7 +55,12 @@ function getFizzBuzz(num) {
  *   10 => 3628800
  */
 function getFactorial(n) {
-    throw new Error('Not implemented');
+    let result = 1;
+    while(n){
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: its not a good practice to change arguments if its not required. Also it's better to use for-loop when you know the number of step in advance

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -63,7 +77,11 @@ function getFactorial(n) {
  *   -1,1  =>  0  ( = -1 + 0 + 1 )
  */
 function getSumBetweenNumbers(n1, n2) {
-    throw new Error('Not implemented');
+    let result = 0;
+    while (n2 >= n1)
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: it's not a good practice to change arguments if its not required. Also it's better to use for-loop when you know the number of step in advance

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -226,7 +257,8 @@ function reverseString(str) {
  *   34143 => 34143
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+    return `${num}`.split('').reverse().join('');
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: function must return a number

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -296,10 +347,24 @@ function getDigitalRoot(num) {
  *   '{[(<{[]}>)]}' = true 
  */
 function isBracketsBalanced(str) {
-    throw new Error('Not implemented');
+    let stack = [],
+    open  = ['[', '(', '{', '<'],
+    close = [']', ')', '}', '>'];
+    for (var i = 0; i < str.length; i++) {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: do not use var keyword. Use either let or const

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -356,7 +461,14 @@ function timespanToHumanString(startDate, endDate) {
  *    365, 10 => '365'
  */
 function toNaryString(num, n) {
-    throw new Error('Not implemented');
+        var result = '';
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: do not use var

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -356,7 +461,14 @@ function timespanToHumanString(startDate, endDate) {
...
+        num = `${num}`;
+            while (num >= n) {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: fix indentation

File: task/09-functions-n-closures-tasks.js [hunk 7]
```diff
@@ -47,7 +47,7 @@ function getComposition(f,g) {
  *
  */
 function getPowerFunction(exponent) {
-    throw new Error('Not implemented');
+    throw new Error('Not implemented');
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js]: you must not have changes in files that are not realted to the task

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by AlexMokin** [comment]
There must no be any changes in task/09-functions-n-closures-tasks.js for this pr. 
Please provide link to your travis ci

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/09-functions-n-closures-tasks.js (hunk 7 above)
Comment (AlexMokin) [reply task/09-functions-n-closures-tasks.js]: there is nothing funny about not being able to do a clean commit

**Comment by AndreiManya** [comment]
https://travis-ci.org/github/AndreiManya/js-assignments/builds/710269635

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**